File: transpile/giada/pizzapazza/math/Z4SignedValue.js
/**
 * A value with sign
 *
 * @author gianpiero.di.blasi
 */
class Z4SignedValue {

   sign = Z4Sign.RANDOM;

   value = 0;

  /**
   * Returns the sign
   *
   * @return The sign
   */
   getSign() {
    return this.sign;
  }

  /**
   * Sets the sign
   *
   * @param sign The sign
   * @return This Z4SignedValue
   */
   setSign(sign) {
    this.sign = sign;
    return this;
  }

  /**
   * Returns the value
   *
   * @return The value
   */
   geValue() {
    return this.value;
  }

  /**
   * Sets the value
   *
   * @param value The value
   * @return This Z4SignedValue
   */
   setValue(value) {
    this.value = value;
    return this;
  }

  /**
   * Returns the next signed value
   *
   * @return The next signed value
   */
   next() {
    return this.sign.next() * this.value;
  }
}

File: src/pizzapazza/math/Z4SignedValue.java
package pizzapazza.math;

/**
 * A value with sign
 *
 * @author gianpiero.diblasi
 */
public class Z4SignedValue implements Z4Nextable<Double> {

  private final Z4Sign sign;
  private final double value;

  /**
   * Creates the object
   *
   * @param sign The sign
   * @param value The value
   */
  public Z4SignedValue(Z4Sign sign, double value) {
    super();
    this.sign = sign;
    this.value = value;
  }

  /**
   * Returns the sign
   *
   * @return The sign
   */
  public Z4Sign getSign() {
    return this.sign;
  }

  /**
   * Returns the (positive) value
   *
   * @return The (positive) value
   */
  public double getValue() {
    return this.value;
  }

  @Override
  public Double next() {
    return this.sign.next() * this.value;
  }
}
